python/ray/util/lightning_accelerators/horovod_ray_accelerator.py

```diff
@@ -0,0 +1,113 @@
+except (ModuleNotFoundError, ImportError):
+    HOROVOD_AVAILABLE = False
+else:
+    HOROVOD_AVAILABLE = True
+
+
+def get_executable_cls():
+    # Only used for testing purposes, currently.
+    # We need to override this in tests to ensure test path is set correctly.
+    return None
+
+
+class HorovodRayAccelerator(HorovodAccelerator):
+    """Pytorch Lightning Accelerator for Horovod training on a Ray cluster.
+
+    This accelerator is used to manage distributed training on a Ray cluster
+    via the Horovod training framework. Internally, the specified number of
+    Ray actors are launched in the cluster and are configured as part of the
+    Horovod ring. The Pytorch Lightning trainer is instantiated on the
+    driver and sent to each of these training workers where training is
+    executed. The distributed training protocol is handled by Horovod.
+
+    Each training worker is configured to reserve 1 CPU and if 1 GPU if
+    ``use_gpu`` is set to ``True``.
+
+    If using this accelerator, you should run your code like a normal Python
+    script: ``python train.py``, and not with ``horovodrun``.
+
+    Args:
+        num_hosts (int): The number of nodes/machines to execute the job on.
+        num_slots (int): Number of workers to be placed on each machine.
+        use_gpu (bool): Whether to use GPU for allocation. For GPU to be
+            used, you must also set the ``gpus`` arg in your Pytorch Lightning
+            Trainer to a value > 0.
+
+    Example:
+
+        .. code_block:: python
+
+            import pytorch_lightning as ptl
+            from ray.util.lightning_accelerators import HorovodRayAccelerator
+
+            ptl_model = MNISTClassifier(...)
+            # 2 nodes, 4 workers per node, each using 1 CPU and 1 GPU.
+            accelerator = HorovodRayAccelerator(num_hosts=2, num_slots=4,
+                use_gpu=True).
+
+            # If using GPUs, set the ``gpus`` arg to a value > 0.
+            # The actual number of GPUs is determined by ``num_slots``.
+            trainer = pl.Trainer(..., gpus=1, accelerator=accelerator).
+            trainer.fit(ptl_model).
+
+    """
+    def __init__(self,
+                 *args,
+                 num_hosts=1,
+                 num_slots=1,
+                 use_gpu=False,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.nickname = "horovod_ray"
+        self.num_hosts = num_hosts
+        self.num_slots = num_slots
+        self.use_gpu = use_gpu
+
+    def setup(self, model):
+        self.trainer.use_horovod = True
+        settings = RayExecutor.create_settings(timeout_s=30)
+        self.executor = RayExecutor(
+            settings,
+            num_hosts=self.num_hosts,
+            num_slots=self.num_slots,
+            use_gpu=self.use_gpu)
+        self.trainer.model = model
+        self.executor.start(executable_cls=get_executable_cls())
+
+    def train(self):
+        results = self.executor.run(self.train_remote)
```

Comment:
Hmm, just as a note, I think this will actually not work for any model larger than 4GB because we end up serializing the whole state. One possibility is to dereference the pointer to `self.model` and then use the Object store to move it over. 

I wouldn't block this PR on that though

Reply:
Oh that's a good point. Is that how it's currently being handled with standard Horovod+Ray?

Reply:
In standard horovod/ray, the model is defined + instantiated within the training function. However, in PTL, the model is expected to be instantiated before the training function is serialized.

Reply:
Ah that's right, got it.

Reply:
This should be supported now- the entire trainer is put into the object store and then fetched later by each worker.